controllers/cluster_controller_test.go

```diff
@@ -2535,6 +2535,60 @@ var _ = Describe("cluster_controller", func() {
 			})
 		})
 
+		Context("mixing storage engine and cluster version", func() {
+			Context("with rocksdb-v1 engine", func() {
+				BeforeEach(func() {
+					generationGap = 0
+				})
+				It("should fail with 6.3.26", func() {
+					cluster.Spec.DatabaseConfiguration.StorageEngine = "ssd-rocksdb-v1"
+					cluster.Spec.Version = "6.3.26"
+					err := k8sClient.Update(context.TODO(), cluster)
+					Expect(err).NotTo(HaveOccurred())
+					_, err = reconcileCluster(cluster)
+					Expect(err).To(HaveOccurred())
+				})
```

Comment:
We already have the reconcile logic here: https://github.com/FoundationDB/fdb-kubernetes-operator/blob/main/controllers/cluster_controller_test.go#L133-L149 that will be always executed. I think the following changes should work:

```
		When("using FDB 6.3.26", func() {
			BeforeEach(func() {
				cluster.Spec.DatabaseConfiguration.StorageEngine = "ssd-rocksdb-v1"
				cluster.Spec.Version = "6.3.26"
				err := k8sClient.Update(context.TODO(), cluster)
				Expect(err).NotTo(HaveOccurred())
				shouldCompleteReconciliation = false
			})

		        // It check that generations are not matching.
		})
```